composer/trainer/trainer.py

```diff
@@ -945,21 +908,23 @@ def __init__(
         # Callbacks
         self.state.callbacks[:] = list(cast(List[Callback], loggers)) + self.state.callbacks
 
+        checkpoint_callbacks = [callback for callback in self.state.callbacks if isinstance(callback, CheckpointSaver)]
         # Checkpoint Saving
+
         self._checkpoint_saver = None
+        # If the user specifies Checkpointing arguments in the Trainer, then create
+        # a _checkpoint_saver callback from their args.
         if checkpoint_save_path is not None:
             self._checkpoint_saver = CheckpointSaver(
                 checkpoint_save_path=checkpoint_save_path,
-                checkpoint_filename=save_filename,
-                artifact_name=save_artifact_name,
-                latest_checkpoint_filename=save_latest_filename,
-                latest_artifact_name=save_latest_artifact_name,
-                overwrite=save_overwrite,
-                weights_only=save_weights_only,
                 checkpoint_save_interval=checkpoint_save_interval,
                 num_checkpoints_to_keep=num_checkpoints_to_keep,
             )
             self.state.callbacks.append(self._checkpoint_saver)
+        # If the user specified a checkpoint callback, then use that as the _checkpoint_saver callback.
+        # TODO(eracah): deal with multiple CheckpointSaver callbacks.
+        elif checkpoint_callbacks:
+            self._checkpoint_saver = checkpoint_callbacks[0]
```

Comment:
For now, let's assume autoresume is happening with the default CheckpointSaver created from Trainer args, and then you can file a JIRA and assign it to me about fixing autoresume. I don't want to overcomplicate this PR for now, and autoresume isn't something we really support with cloud rn.

Basically, this fixes a scenario we don't address right now, and I'm fine leaving it unaddressed and fixing in a separate PR. Also, I can handle it because I imagine we'll rework parts of autoresume once we have the cloud support anyways

Reply:
Ok that's fine. I need this logic for the next PR, so I will keep it.

Reply:
assigned you a JIRA!